### PR TITLE
Clean house, and add a binary that runs the generator.

### PR DESCRIPTION
**Pull request by aomarks**
Just getting familiar here. This PR cleans house, adds boilerplate files, sets up Travis (just to verify build for now, no tests yet), and adds a binary that by default runs the generator on the CWD and dumps the result to stdout.

### CODE REVIEW

File: src/cli.ts [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    process.stdout.write(declarations);
+  }
+}
+
+(async () => {
+  try {
+    await run(process.argv);
+  } catch (e) {
+    console.error(e);
+    process.exit(64);
```
Comment (justinfagnani) [inline src/cli.ts]: what's 64? Add comment

File: src/gen-ts.ts [hunk 2]
```diff
@@ -60,12 +63,14 @@ function generatePackage(pkg: Analysis): Promise<string> {
   return Promise.resolve(declarations.join('\n'));
 }
 
-function genElementDeclaration(element: Element, declarations: string[], indent: number = 0) {
+function genElementDeclaration(
+    element: Element, declarations: string[], indent: number = 0) {
   if (!element.className) {
     // TODO: handle elements with tagnae, but no exported class
```
Comment (justinfagnani) [inline src/gen-ts.ts:69]: console warn (and typo)

File: src/gen-ts.ts [hunk 3]
```diff
@@ -118,45 +127,50 @@ function genElementDeclaration(element: Element, declarations: string[], indent:
...
+  for (const [, method] of mixin.methods) {
     const methodText = genMethod(method, indent + 2);
     if (methodText) {
       d += methodText;
     }
   }
 
   d += `${idt(indent)}}\n`;
-  d += `${idt(indent)}const ${name}: <T extends Constructor<HTMLElement>>(base: T) => T & Constructor<${name}>;\n`;
+  d += `${idt(indent)}const ${
```
Comment (justinfagnani) [inline src/gen-ts.ts]: unfortunate formatting

**Review by justinfagnani — APPROVED** [review]

File: src/gen-ts.ts (hunk 3 above)
Comment (aomarks) [reply src/gen-ts.ts]: Fixed.

File: src/cli.ts (hunk 1 above)
Comment (aomarks) [reply src/cli.ts]: It was based on the recommendation in http://tldp.org/LDP/abs/html/exitcodes.html#EXITCODESREF to use 64-113 for user-defined error codes. But I don't think this actually has any practical benefit (it's pretty unusual to use specific exit code values for debugging), so I'll change it to 1 to be less confusing.

File: src/gen-ts.ts (hunk 2 above)
Comment (aomarks) [reply src/gen-ts.ts:69]: I'll fix this up later, going to go through all the code next anyway.

**Pull request merged**